core/src/main/java/org/infinispan/protostream/TagHandler.java

```diff
@@ -13,8 +15,10 @@
 
    /**
     * Start of top level message. Do whatever required init here.
+    *
+    * @param descriptor The {@link GenericDescriptor} for the top level Enum or Message. For primitives it is null.
     */
-   default void onStart() {
+   default void onStart(AnnotatedDescriptor descriptor) {
```

Comment:
I've added this in order to have access to the full type name in the beginning of the stream in order to render the ```_type``` field in the JSON message. Do you see any issue with backwards compat? Protostream is not usually a user face library, and to migrate ISPN it's necessary to replace ```onStart()``` to ```onStart(null)``` in 2 or 3 places